### PR TITLE
Fixed 3.0 amqp_interop driver and removed deprecated amqp driver

### PR DESCRIPTION
**Pull request by kids-return**
Fixed 3.0 amqp_interop driver and removed deprecated amqp driver
config/common.php
```php
    'queue' => [
        '__class' => \yii\queue\amqp_interop\Queue::class,
        'host' => '192.168.10.20',
        'port' => 5672,
        'user' => 'guest',
        'password' => 'guest',
        'queueName' => 'queue',
        'driver' => \yii\queue\amqp_interop\Queue::ENQUEUE_AMQP_LIB,
    ],
```

### CODE REVIEW

File: src/drivers/amqp_interop/Queue.php [hunk 1]
```diff
@@ -235,7 +237,8 @@ public function listen()
 
         $queue = $this->context->createQueue($this->queueName);
         $consumer = $this->context->createConsumer($queue);
-        $this->context->subscribe($consumer, function (AmqpMessage $message, AmqpConsumer $consumer) {
+        $subscriptionConsumer = $this->context->createSubscriptionConsumer();
+        $subscriptionConsumer->subscribe($consumer, function (AmqpMessage $message, AmqpConsumer $consumer) {
```
Comment (kids-return) [inline src/drivers/amqp_interop/Queue.php]: `"enqueue/amqp-lib": "^0.9"` Need to use this way

**Comment by samdark** [comment]
Can we do it two separate pull requests? Bug fix could be released in a minor version while removal of a driver requires major one.

**Comment by machour** [comment]
@samdark This is yii-queue, not yii2-queue 🤔

**Comment by kids-return** [comment]
https://github.com/yiisoft/yii-queue/blob/31c433c30c0c6a382d0d03386378f3449885e0e6/src/drivers/amqp/Queue.php#L21
Planned to be removed in version 3.0

**Comment by samdark** [comment]
Right right. Haven't noticed "2" in the URL. Sorry.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**